plugin/store/store.go

```diff
@@ -98,12 +98,31 @@ func (ps *Store) SetState(p *v2.Plugin, state bool) {
 }
 
 // Add adds a plugin to memory and plugindb.
-func (ps *Store) Add(p *v2.Plugin) {
+// An error will be returned if there is a collision.
```

Comment:
Callers of `pluginStore.Add` in backend_linux.go (Pull) and manager.go (init) should also be updated to check for returned error.